tests/test_downloadermiddleware.py

```diff
@@ -176,3 +177,31 @@ def process_exception(self, request, exception, spider):
         dfd.addBoth(results.append)
         self.assertIsInstance(results[0], Failure)
         self.assertIsInstance(results[0].value, _InvalidOutput)
+
+
+class MiddlewareUsingDeferreds(ManagerTestCase):
+    """Middlewares using Deferreds should work"""
```

Comment:
This is a strictly unrelated change. It just shows that deferred middleware methods work and that they won;t break when adding coroutine method support.